src/coreclr/jit/targetamd64.h

```diff
@@ -92,7 +92,9 @@
   #define REG_MASK_FIRST           REG_K0
   #define REG_MASK_LAST            REG_K7
 
-  #define RBM_ALLMASK              RBM_K1
+  #define RBM_ALLMASK_INIT         (0)
+  #define RBM_ALLMASK_EVEX         (RBM_K1 | RBM_K2 | RBM_K3 | RBM_K4 | RBM_K5 | RBM_K6 | RBM_K7)
```

Comment:
Notably this doesn't include `K0` because it is a bit special.

In particular, while `K0` can be used as an input or an output of many of the `kmask instructions`, it cannot be used as a predicate to many of the SIMD instructions because it instead represents "no predication".

We should likely ensure `K0` can still be used longer term, but for this PR it's unnecessary.